compiler/bindings/python/iree/build/executor.py

```diff
@@ -289,20 +298,43 @@ def __init__(
     ):
         super().__init__(executor=executor, deps=deps)
         self.desc = desc
-        self.concurrnecy = concurrency
+        self.concurrency = concurrency
 
     def __str__(self):
         return self.desc
 
     def __repr__(self):
         return f"Action[{type(self).__name__}]('{self.desc}')"
 
-    def invoke(self):
-        self._invoke()
+    def invoke(self, scheduler: "Scheduler"):
+        # Invoke is run within whatever in-process execution context was requested:
+        #   - On the scheduler thread for NONE
+        #   - On a worker thread for THREAD or PROCESS
+        # For PROCESS concurrency, we have to create a compatible invocation
+        # thunk, schedule that on the process pool and wait for it.
+        if self.concurrency == ActionConcurrency.PROCESS:
+            thunk = self._remotable_thunk()
+            fut = scheduler.process_pool_executor.submit(thunk)
+            fut.result()
```

Comment:
Well I was going to suggest not abbreviating `future` to `fut`, but I imagine this is a somewhat common naming, and it avoids potential name conflicts like `arr` instead of `array`?

https://www.reddit.com/r/Python/comments/r6jz55/what_naming_convention_do_you_prefer_for_a_list/ I like the suggestion there
> Do not name it as the container type but as what it is. ‘apples’ instead ‘list’ (of apples). When it is too abstract and I don’t find a name, I use ‘alist’ or, more conventional, ‘list_’. 

but here the future is pretty generic.

Whatever :D

Reply:
And very contained. And what it is is a future only signaling completion. So leaving it.